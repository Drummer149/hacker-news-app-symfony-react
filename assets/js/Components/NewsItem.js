import React from 'react';
import { getHostname, timeSince} from '../Common/Utils';

// Populates a NewsItem from a news object
const NewsItem = ({ obj }) => (
    <div className="newsItem">
      <div className="section">
        <span className="title"><a href={obj.url}>{obj.title}</a></span>
        <span className="websiteName"> {getHostname(obj.url)}</span>
      </div>
      <div className="section sublinks">
        <span>{obj.score} points</span>
        <span> by {obj.by}</span>
        <a href={`/item?id=${obj.id}`}><span> {timeSince(obj.time)} ago</span></a>
        <a href={`/item?id=${obj.id}`}><span> {obj.descendants} Comments</span></a>
      </div>
    </div>
);

export default NewsItem;