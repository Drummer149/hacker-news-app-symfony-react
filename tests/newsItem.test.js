import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
configure({ adapter: new Adapter() });

import React from 'react';
import { shallow } from 'enzyme';
import NewsItem from '../assets/js/Components/NewsItem';

describe('News Item', () => {
  const initialProps = {
    obj: {
      url: '',
      title: '',
      descendants: 0,
      id: 0,
      time: 10000000,
    },
    index: 0
  }

  it('should render correctly in "debug" mode', () => {
    const component = shallow(<NewsItem {...initialProps} />);
  
    expect(component).toMatchSnapshot();
  });
});