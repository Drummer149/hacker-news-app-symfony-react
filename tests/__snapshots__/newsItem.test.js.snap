// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`News Item should render correctly in "debug" mode 1`] = `ShallowWrapper {}`;
